packages/gatsby-source-contentful/src/plugin-options.js

```diff
@@ -0,0 +1,40 @@
+const Joi = require(`joi`)
+
+const defaultOptions = {
+  host: `cdn.contentful.com`,
+
+  // default is always master
+  environment: `master`,
+}
+
+const optionsSchema = Joi.object().keys({
+  accessToken: Joi.string()
+    .required()
+    .empty(),
+  spaceId: Joi.string()
+    .required()
+    .empty(),
+  host: Joi.string()
```

Comment:
I think host can be optional here. if nothing is passed to the SDK will assume it's `cdn.contentful.com`

Reply:
It is in practice optional, because I merge defaultOptions with passed options before validating - but I probably should shuffle code around so the definition of shape of options is more clear

Reply:
so I will look into doing that

Reply:
Addressed after refactoring